### PR TITLE
Docker fix

### PR DESCRIPTION
**Pull request by stukalin**
**Changes proposed in this pull request:**
- fix Dockerfile since alpine doesn't have many required packages
- enable autoreloading via docker-compose
- readme fix

Somehow #1300 brought issues in Docker container building which were not identified right away (very-very strange). Proposed is the fix for that.

cc @HospitalRun/core-maintainers

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,34 @@
...
+  curl \
+  g++ \
+  git \
+  make \
+  python \
+  sudo \
+  wget
+
+# install global npm dependencies
+RUN npm install -g ember-cli@latest && npm install -g bower
```
Comment (stukalin) [inline Dockerfile]: Here I've slightly changes the order of commands in order to use docker cache

File: Dockerfile [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
+
+# use changes to package.json to force Docker not to use the cache
+# when we change our application's nodejs dependencies:
+COPY package*.json /tmp/
+RUN cd /tmp && npm install
+RUN mkdir -p /usr/src/app && cp -a /tmp/node_modules /usr/src/app
+
+COPY bower.json /tmp/
```
Comment (stukalin) [inline Dockerfile]: bower goes here

File: README.md [hunk 3]
```diff
@@ -80,7 +80,7 @@ To run HospitalRun with [Docker](https://www.docker.com/) please do the followin
 - Clone the repository with the command `git clone https://github.com/HospitalRun/hospitalrun-frontend.git`.
 - Change to the hospitalrun-frontend directory `cd hospitalrun-frontend`.
 - Build the HospitalRun image with `docker build -t hospitalrun-frontend .`.
-- Execute `docker run -it --name couchdb -d couchdb` to create the couchdb container.
+- Execute `docker run -it --name couchdb -d couchdb:1.7.1` to create the couchdb container.
```
Comment (stukalin) [inline README.md]: Couchdb 1.* is our primary

File: Dockerfile (hunk 2 above)
Comment (DerekTBrown) [reply Dockerfile]: you don't need this

Comment (stukalin) [reply Dockerfile]: Why not? Saves several seconds time on rebuild. I remember that we're gonna get rid of bower but we haven't yet

**Pull request merged**